test/files/scalacheck/MutableTreeMap.scala

```diff
@@ -0,0 +1,249 @@
+import java.io._
+
+import org.scalacheck._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop.forAll
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.util.Try
+
+trait MutableTreeMapGenerators {
+
+  // Note: in scalacheck 1.12.2 tree maps can be automatically generated without the need for custom
+  // machinery
+  def genTreeMap[A: Arbitrary: Ordering, B: Arbitrary]: Gen[mutable.TreeMap[A, B]] = {
+    import org.scalacheck.Gen._
+    for {
+      keys <- listOf(arbitrary[A])
+      values <- listOfN(keys.size, arbitrary[B])
+    } yield mutable.TreeMap(keys zip values: _*)
+  }
+
+  implicit def arbTreeMap[A: Arbitrary: Ordering, B: Arbitrary] = Arbitrary(genTreeMap[A, B])
+}
+
+object MutableTreeMapProperties extends Properties("mutable.TreeMap") with MutableTreeMapGenerators {
```

Comment:
Very nice set of checks!

However, what's missing here is checking the tree invariants. Could you add an assertion/property in each of the checks that verifies that each of the produced trees has the correct invariants?